students/johnpharmd/lesson09/mailroom_OO.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+import sys
+
+
+class Donor:
+    """creates objects for individual donors"""
+    def __init__(self, title, last_name, donation):
+        # fixme: self.ID = 0
+        self.title = title
+        self.last_name = last_name
+        self.donations = [donation]
+        super().__init__()
+
+    @property
+    def donor(self):
+        """getter for individual donor"""
+        return {self.last_name: {'title': self.title, 'donations':
+                sum(self.donations), 'num_donations': len(self.donations)}}
+
+    @property
+    def title(self):
+        return self.new_title
+
+    @title.setter
+    def title(self, new_title):
+        """enables title change for donor"""
+        self.new_title = new_title
+
+    @property
+    def donation(self):
+        """getter for most recent donation amount"""
+        return self.donations[-1]
+
+    @donation.setter
+    def donation(self, donation):
+        """enables addition of new donation"""
+        self.donations.append(donation)
+
+
+class DonorGroup(Donor):
```

Comment:
this class shouldn't extend from `Donor`

`Donor` represents a single donor object (which you have defined very well above)
`DonorGroup` should be a collection of donor objects and any action on that. Think of Dog is a Pet relationship that makes sense for inheritance, but DonorGroup is Donor doesn't really make sense.